File: Deepseek-Coder-33b-Instruct/stage2/atomic_construct.cpp
#include <iostream>
#include <openacc.h>

#ifndef T1
//T1:atomic construct,V:2.7-3.3
int test1(){
    int err = 0;
    int shared_var = 0;

    #pragma acc parallel loop copy(shared_var)
    for (int i = 0; i < 1000; i++) {
        #pragma acc atomic update
        shared_var++;
    }

    if (shared_var != 1000) {
        err = 1;
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < 1; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Deepseek-Coder-33b-Instruct/stage2/atomic_construct.f90
!T1:atomic construct,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, PARAMETER :: N = 1000
        INTEGER :: i, v
        INTEGER, DIMENSION(:), ALLOCATABLE :: a

        ALLOCATE(a(N))

        !$acc data copy(a)
        !$acc parallel loop
        DO i = 1, N
          a(i) = i
        END DO
        !$acc end parallel loop
        !$acc end data

        v = 0
        !$acc atomic update
        v = v + 1
        !$acc end atomic

        !$acc parallel loop
        DO i = 1, N
          !$acc atomic update
          a(i) = a(i) + v
          !$acc end atomic
        END DO
        !$acc end parallel loop

        !$acc parallel loop
        DO i = 1, N
          IF (a(i) .NE. i + v) THEN
            errors = errors + 1
          END IF
        END DO
        !$acc end parallel loop

        DEALLOCATE(a)

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
